File: storage/rdb/src/main/java/project/dailyge/entity/codeandmessage/CodeAndMessageJpaEntity.java
package project.dailyge.entity.codeandmessage;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Builder;
import lombok.Getter;
import project.dailyge.entity.BaseEntity;

@Getter
@Entity(name = "code_and_message")
public class CodeAndMessageJpaEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "domain")
    private String domain;

    @Column(name = "name")
    private String name;

    @Column(name = "code")
    private int code;

    @Column(name = "message")
    private String message;

    protected CodeAndMessageJpaEntity() {
    }

    private CodeAndMessageJpaEntity(
        final String domain,
        final String name,
        final int code,
        final String message
    ) {
        this.domain = domain;
        this.name = name;
        this.code = code;
        this.message = message;
    }

    @Builder
    public static CodeAndMessageJpaEntity create(
        final String domain,
        final String name,
        final int code,
        final String message
    ) {
        return new CodeAndMessageJpaEntity(domain, name, code, message);
    }

    public void update(final CodeAndMessageJpaEntity target) {
        this.domain = target.getDomain();
        this.name = target.getName();
        this.code = target.getCode();
        this.message = target.getMessage();
    }
}

File: storage/rdb/src/main/kotlin/project/dailyge/entity/codeandmessage/CodeAndMessageJpaEntity.kt
package project.dailyge.entity.codeandmessage

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import project.dailyge.entity.BaseEntity

@Entity(name = "code_and_message")
class CodeAndMessageJpaEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "domain")
    private var _domain: String,

    @Column(name = "name")
    private var _name: String,

    @Column(name = "code")
    private var _code: Int,

    @Column(name = "message")
    private var _message: String,
) : BaseEntity() {

    companion object {
        fun create(
            domain: String,
            name: String,
            code: Int,
            message: String,
        ): CodeAndMessageJpaEntity {
            return CodeAndMessageJpaEntity(_domain = domain, _name = name, _code = code, _message = message)
        }
    }

    val domain: String
        get() = _domain

    val name: String
        get() = _name

    val code: Int
        get() = _code

    val message: String
        get() = _message

    fun update(target: CodeAndMessageJpaEntity) {
        _domain = target.domain
        _name = target.name
        _code = target.code
        _message = target.message
    }
}
